### ISSUE TITLE
New UI next steps (?)

### ISSUE THREAD

**Issue by DanVanAtta** [triplea-game/triplea#2753]
New UI is now available behind a beta flag, not fully functional yet - a question here for dev team here on how to proceed.  We have a few options:
1. Try to make new UI work with existing network model code. 
2. Remove existing UI so we can refactor model code and then integrate it with the new UI
3. Build a new network code infrastructure (JSON over Java sockets or something similar)

Option number one is a bit painful, but with enough stamina and teeth gritting and hacking we could get it to work. A nice bonus is that we can have both UIs, new and old side by side with both working.

Option two would make integration of network model code with the new UI much easier. We would also be able to drop a very large amount of 'legacy' UI code.

Option three requires potentially the most work as we would be rebuilding the network model, and this could have implications for actual gameplay as well beyond the staging screens. If we want to start a new network model/codebase, this could be a good option. A downside to this approach is that it would probably take longest as we rewrite TripleA's network code.

So question for dev's, which option seems most reasonable/best? I think I am leaning towards number one or two, but if there is some help I could be talked into number three. Number two is a bit attractive, but it requires us to commit with going with the new UI and not turning back. Happily the new UI does have all the screens mostly completed, largely we just need to wire together the network models so that the game launches (so it is actually pretty close). IF we drop the old UI code and can then refactor the network model code, that would likely be the most efficient and interestingly least risky for getting the new UI launched. On the other hand there is some good value in having both UI's running in parallel while we work to complete the UI update.

**Comment by RoiEXLab** [comment]
@DanVanAtta Option 2 would probably a good compromise.
The network code really needs some work in general. Currently there's no way to implement a new feature without it causing an exception until the lobby (or the other client) is being updated.
Also the current state of the network code doesn't look like it can easily be changed to support TLS encryption.

TL;DR: I'd prefer option 2, especially because this would make JavaFX easier as well. The JavaFX UI will probably take longer than the "new swing UI". The biggest difference being that every UI component is going to be initialized exactly once and being displayed when neccessary, all in one window to allow for fullscreen etc. (exception: FileChooser/DirectoryChooser dialogs)

**Comment by DanVanAtta** [comment]
Thanks for the feedback - a side note @RoiEXLab  - My hope is that the 'new UI', using the custom Swing builder lib's will make JavaFX migration easier. The builders should help us split the logic from the UI code (and UI configuration), so eventually, when we go to JavaFX the builder API stuff could maybe be cleanly replaced.

**Comment by RoiEXLab** [comment]
@DanVanAtta I hope so as well.

One thing I'm probably going to submit a PR for soon is the way the current UI handles the ClientSettings:
Currently some ClientSettings have a "not set" state, where they default to the hardcoded value. Instead they should have a "default" value that gets restored when klicking on "reset to default".
Little things like this already make implementing a new UI easier.

**Comment by RoiEXLab** [comment]
@DanVanAtta But at some point it already did: The ClientSetting "framework" helped a lot when implementing the settings in JavaFX
